src/comm/Incoming.h

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID__SRC_COMM_INCOMING_H
+#define SQUID__SRC_COMM_INCOMING_H
+
+#if USE_POLL || USE_SELECT
+
+#include "SquidConfig.h"
+#include "StatHist.h"
+
+namespace Comm
+{
+
+/**
+ * Automatic tuning for incoming requests.
+ *
+ * INCOMING sockets are the listening ports for transport protocols.
+ * We need to check these fairly regularly, but how often?  When the
+ * load increases, we want to check the incoming sockets more often.
+ * If we have a lot of one protocol incoming, then we need to check
+ * those sockets more than others.
+ *
+ * \copydoc Comm::Incoming::check()
+ *
+ * \copydoc Comm::Incoming::finishPolling()
+ *
+ * Caveats:
+ *
+ *   \copydoc INCOMING_FACTOR
+ *
+ *   \copydoc Comm::Incoming::MaxInterval
+ */
+class Incoming
+{
+public:
+#if !defined(INCOMING_FACTOR)
```

Comment:
Would it be possible to use a const integer type instead, and move the preprocessor-based default value setting outside the class?

Reply:
Done what can be done in [e2da9e8](https://github.com/squid-cache/squid/pull/1572/commits/e2da9e8d68417728303809b330d9873db8f7f7af). The `#ifdef` is inside the class to make it clear that the magic value is only relevant in that scope.